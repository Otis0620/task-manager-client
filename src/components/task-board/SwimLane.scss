.swimLane {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
  width: 25%;
  padding: 20px;
  border: 0.5px solid lightgray;
  margin: 10px;
  box-shadow: 8px 9px 11px 0px rgba(197, 197, 197, 1);
}

.swimLane-title {
  display: flex;
  justify-content: space-around;
  border-radius: 2px;
}

.swimLane-title p {
  color: white;
}