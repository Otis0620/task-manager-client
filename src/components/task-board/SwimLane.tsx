import './SwimLane.css';
import React, { FC } from 'react';
import { SwimLanePropsI } from './interfaces';

export const SwimLane: FC<SwimLanePropsI> = ({ backgroundColor, swimLaneTitle, children }) => (
  <div className={`swimLane`}>
    <div className="swimLane-title" style={{ backgroundColor: backgroundColor }}>
      <p>{swimLaneTitle}</p>
    </div>

    {children}
  </div>
);
