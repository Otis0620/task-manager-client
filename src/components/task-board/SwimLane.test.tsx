import React from 'react';
import { shallow } from 'enzyme';
import { SwimLane } from './SwimLane';

test('SwimLane', () => {
  const wrapper = shallow(<SwimLane backgroundColor="red" swimLaneTitle="Backlog" />);

  expect(wrapper).toMatchSnapshot();
});
