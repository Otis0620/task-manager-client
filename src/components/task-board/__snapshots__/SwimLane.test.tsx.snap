// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwimLane 1`] = `
<div
  className="swimLane"
>
  <div
    className="swimLane-title"
    style={
      Object {
        "backgroundColor": "red",
      }
    }
  >
    <p>
      Backlog
    </p>
  </div>
</div>
`;
